pkg/manager/member/tikv_scaler.go

```diff
@@ -144,7 +173,7 @@ func (tsd *tikvScaler) ScaleIn(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSe
 			}
 			klog.Infof("tikv scale in: set pvc %s/%s annotation: %s to %s",
 				ns, pvcName, label.AnnPVCDeferDeleting, now)
-
+			tc.Status.TiKV.Phase = v1alpha1.ScaleInPhase
```

Comment:
I think we can't directly set TiKV phase as `ScaleInPhase` here. During scaling, we would update its `statefulset` and `tidbcluster`. What if  the updating of statefulset is success but the updating of tidbcluster is failed? This would cause the scaling truly happened, but we didn't record the state.

I think we should judge the status in `syncTidbClusterStatus` function as upgrading do.

I

Reply:
We cannot exactly know whether TiKV is scaling in the `syncTidbClusterStatus` function, because the operator may be deleting the store at the time we're checking and we also cannot check the store state because if a store is in tombstone state we cannot make sure it's the store for the current Pod and or a previous Pod that had been deleted before.
We have a retry mechanism for the tidbcluster update and yes, it still can fail, but actually we cannot fix all of the corner cases, this PR will mitigate this issue in most cases.